src/devices/Card/Icode/IcodeCard.cs

```diff
@@ -0,0 +1,419 @@
+
+        /// <summary>
+        /// AFI (Application family identifier) represents the type of application targeted by the VCD
+        /// </summary>
+        public byte Afi { get; set; }
+
+        /// <summary>
+        /// The Data storage format identifier indicates how the data is structured in the VICC memory
+        /// </summary>
+        public byte Dsfid { get; set; }
+
+        /// <summary>
+        /// Constructor for IcodeCard
+        /// </summary>
+        /// <param name="rfid">A card transceiver class</param>
+        /// <param name="target">The target number as some card readers attribute one</param>
+        public IcodeCard(CardTransceiver rfid, byte target)
+        {
+            _rfid = rfid;
+            Target = target;
+            _logger = this.GetCurrentClassLogger();
+        }
+
+        /// <summary>
+        /// Provide a calculation of CRC for ISO15693
+        /// The PN5180 module seems to have implemented crc and does not need to calculate when coding
+        /// </summary>
+        /// <param name="buffer">The buffer to process</param>
+        /// <param name="crc">The CRC, Must be a 2 bytes buffer</param>
+        public void CalculateCrcIso15693(ReadOnlySpan<byte> buffer, Span<byte> crc)
+        {
+            if (crc.Length != 2)
+            {
+                throw new ArgumentException($"The length of crc must be 2 bytes.", nameof(crc));
+            }
+
+            ushort polynomial = 0x8408;
+            ushort currentCrc = 0xFFFF;
+            // ISO15693-3.pdf
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                currentCrc = (ushort)(currentCrc ^ buffer[i]);
+                for (int j = 0; j < 8; j++)
+                {
+                    if ((currentCrc & 0x0001) != 0)
+                    {
+                        currentCrc = (ushort)((currentCrc >> 1) ^ polynomial);
+                    }
+                    else
+                    {
+                        currentCrc = (ushort)(currentCrc >> 1);
+                    }
+                }
+            }
+
+            currentCrc = (ushort)~currentCrc;
+            crc[0] = (byte)(currentCrc & 0xFF);
+            crc[1] = (byte)((currentCrc >> 8) & 0xFF);
+        }
+
+        /// <summary>
+        /// Run the last setup command. In case of reading bytes, they are automatically pushed into the Data property
+        /// </summary>
+        /// <returns>-1 if the process fails otherwise the number of bytes read</returns>
+        private int RunIcodeCardCommand()
+        {
+            byte[] requestData = Serialize();
+            byte[] dataOut = new byte[_responseSize];
+
+            var ret = _rfid.Transceive(Target, requestData, dataOut.AsSpan(), NfcProtocol.Iso15693);
+            _logger.LogDebug($"{nameof(RunIcodeCardCommand)}: {_command}, Target: {Target}, Data: {BitConverter.ToString(requestData)}, Success: {ret}, Dataout: {BitConverter.ToString(dataOut)}");
```

Comment:
I know it's minimal but it can have some performances impact. The `$"wih everything inside"` expression is always calculated then the LogDebug function is called and inside, there is a check about the log level saying that the log should be displayed or not.
Point here is that the creation of the expression, debug or not is always processed.
So, can you please check the log level and then call the LogDebug is we're indeed in debug mode?